### ISSUE TITLE
insufficient input error in combination with quoted vector

### ISSUE THREAD

**Issue by mk** [clj-kondo/clj-kondo#2580]
**version**

clj-kondo v2025.07.26

**platform**

native version from brew

**problem**

Getting a false positive error

**repro**

Minimal repro
```sh
$ echo "(assoc {} :foo '[bar] :id #{})"  | clj-kondo --lint -
<stdin>:1:27: error: Insufficient input.
linting took 7ms, errors: 1, warnings: 0

$ clj-kondo --version
```

**expected behavior**

No error
